[refman] Add link to the deprecation policy (#17532)
coq/coq: doc/sphinx/using/libraries/writing.rst

      :n:`@string__since` is the version number, :n:`@string__note` is
      the note (usually explains the replacement).

.. note::

>    Coq itself and its standard library apply the following deprecation policy:

Reviewer: ```suggestion
   Coq and its standard library follow this deprecation policy:
```
